lib/awful/util.lua

```diff
@@ -512,7 +512,7 @@ function util.table.iterate(t, filter, start)
     return function ()
         while count < length do
             local item = t[index]
-            index = util.cycle(#t, index + 1)
+            index = gmath.cycle(#t, index + 1)
```

Comment:
But if I merge your PR first, it will breaks the CI. Right now the CI doesn't care about "future deprecation". So this one is safe to merge now. But I can't merge your PR without breaking the CI. I propose merging this PR now. It will be less work for you, not more.